src/main/scala/mesosphere/marathon/core/volume/providers/DVDIProvider.scala

```diff
@@ -0,0 +1,247 @@
+package mesosphere.marathon.core.volume.providers
+
+import com.wix.accord.{ Validator, _ }
+import com.wix.accord.dsl._
+import mesosphere.marathon.core.volume._
+import mesosphere.marathon.state._
+import org.apache.mesos.Protos.{ ContainerInfo, Environment, Volume => MesosVolume }
+
+import scala.collection.JavaConverters._
+
+/**
+  * DVDIProvider (Docker Volume Driver Interface provider) handles persistent volumes allocated
+  * by a specific docker volume driver plugin. This works for both docker and mesos containerizers,
+  * albeit with some limitations:
+  *   - only a single volume driver per container is allowed when using the docker containerizer
+  *   - docker containerizer requires that referenced volumes be created prior to application launch
+  *   - mesos containerizer only supports volumes mounted in RW mode
+  */
+protected[volume] case object DVDIProvider
+    extends AbstractPersistentVolumeProvider("external") {
+
+  import org.apache.mesos.Protos.Volume.Mode
+  import OptionSupport._
+
+  abstract class ProviderOption(
+    name: String,
+    required: Boolean = false,
+    validValue: Validator[String] = labelValidator) extends NamedOption(
+    DVDIProvider.this.name, name, required, validValue)
+
+  case object OptionDriverName extends ProviderOption("driver", true)
```

Comment:
Moreover, if the driver is not given as an option, we get very strange violations:
```
{"message":"Object is not valid","details":[{"path":"/value","errors":["does not equal 1"]},{"path":"/container/volumes(0)/persistent.options","errors":["is a required option, but is not present"]},{"path":"/value/container.type","errors":["does not equal MESOS"]}]}
```